runtime/runtime/src/lib.rs

```diff
@@ -1331,6 +1351,11 @@ impl Runtime {
             }
         }
 
+        // No more receipts are executed on this trie, stop any pending prefetches on it.
+        if let Some(prefetcher) = &prefetcher {
+            prefetcher.stop_prefetching();
```

Comment:
This should maybe be part of the `impl Drop for TriePrefetcher`

Reply:
That's already what the current `impl Drop for TriePrefetcher` does effectively. It is required anyway for all failure cases that may happen above.

Would you like to see `drop(prefetcher)` here instead of calling `stop_preftching`?

Reply:
Hmm I think it would be dropped by end-of-scope anyway? (unless the code changes, I'l re-mention it in the new if it's important anyway)